stograde/specs/cache.py

```diff
@@ -76,13 +76,40 @@ def convert_spec(yaml_path, json_path):
 def clarify_yaml(data):
     copied = copy.deepcopy(data)
     if 'files' in copied and copied['files'] is not None:
-        copied['files'] = [process_filelike_into_dict(f) for f in copied['files']]
+        copied['files'] = [process_file_yaml_into_dict(f) for f in copied['files']] \
+            if 'new_spec' in copied and copied['new_spec'] is True else \
+            [process_file_yaml_into_dict_legacy(f) for f in copied['files']]
     if 'tests' in copied and copied['tests'] is not None:
-        copied['tests'] = [process_filelike_into_dict(f) for f in copied['tests']]
+        copied['tests'] = [process_file_yaml_into_dict(f) for f in copied['tests']] \
+            if 'new_spec' in copied and copied['new_spec'] is True else \
+            [process_file_yaml_into_dict_legacy(f) for f in copied['tests']]
     return copied
 
 
-def process_filelike_into_dict(file_list):
+def process_file_yaml_into_dict(file_list):
+    filename = file_list['file']
+    if filename is None:
+        raise Exception("File name must be specified")
+
+    commands = [] if 'commands' not in file_list else \
+        [file_list['commands']] if isinstance(file_list['commands'], str) else \
+        file_list['commands'] if isinstance(file_list['commands'], list) else None
+    if commands is None:
+        raise TypeError
```

Comment:
```python
assert 'commands' in file_list, KeyError('expected "commands" key in file_list; got {} keys'.format(file_list.keys()))

commands = file_list['commands']
if isinstance(commands, str):
    commands = [commands]

assert isinstance(commands, list)
```

Reply:
I don't want to throw an error if `commands` or `options` aren't present though. I'll type something up when I'm on my computer again.

Reply:
Ah, I wasn't quite sure of the None behaviors in your approach. I'll take a look at the new one.